builtin/providers/aws/resource_aws_autoscaling_group.go

```diff
@@ -244,7 +244,13 @@ func resourceAwsAutoscalingGroup() *schema.Resource {
 				},
 			},
 
-			"tag": autoscalingTagsSchema(),
+			"tag": autoscalingTagSchema(),
+
+			"tags": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeMap},
+			},
```

Comment:
I see what do you mean - static, predefined tags combined with more tags from a variable.
I think this should work just fine

```hcl
tags = [
    {
      key                 = "foo"
      value               = "bar"
      propagate_at_launch = true
    },
    {
      key                 = "lorem"
      value               = "ipsum"
      propagate_at_launch = false
    },
    "${var.additional_tags}"
  ]
```

The main reason to make those mutually exclusive is to avoid having to document and deal with edge cases like duplicate tags or even conflicting ones (which one is the point of truth) and also to avoid any potential confusion in the heads of users - i.e. I don't want users to have to think about edge cases at all.

Also we don't have an acceptance test for importing ASGs yet, but I'm pretty confident it would fail in this PR because you're assuming that there's a config available - i.e. `GetOk()` in `Read()` decides which field to populate and during import none of those fields will be available, so none will be populated.

We need to decide which field is going to be the point of truth for this context. I'd vote for the existing `tag` field to avoid breaking changes.

I'm going to create the mentioned test and cc you in the PR, just for context.